metals/src/main/scala/scala/meta/metals/lsp/DelegatingScalaService.scala

```diff
@@ -0,0 +1,206 @@
+package scala.meta.metals.lsp
+
+import java.util
+import java.util.concurrent.CompletableFuture
+
+import scala.meta.internal.metals.DidFocusResult
+import scala.meta.internal.metals.HoverExtParams
+import scala.meta.internal.metals.WindowStateDidChangeParams
+import scala.meta.internal.metals.doctor.DoctorVisibilityDidChangeParams
+import scala.meta.internal.metals.findfiles.FindTextInDependencyJarsRequest
+import scala.meta.internal.tvp.MetalsTreeViewChildrenResult
+import scala.meta.internal.tvp.TreeViewChildrenParams
+import scala.meta.internal.tvp.TreeViewNodeCollapseDidChangeParams
+import scala.meta.internal.tvp.TreeViewNodeRevealResult
+import scala.meta.internal.tvp.TreeViewParentParams
+import scala.meta.internal.tvp.TreeViewParentResult
+import scala.meta.internal.tvp.TreeViewVisibilityDidChangeParams
+
+import org.eclipse.lsp4j._
+import org.eclipse.lsp4j.jsonrpc.messages.{Either => JEither}
+
+/**
+ * Delegating Scala LSP service which forwards all requests to the underlying
+ * instance. This is needed to support lsp4j
+ * [[org.eclipse.lsp4j.jsonrpc.services.JsonDelegate]]. Value returned by
+ * method with that annotation is used to determine which methods are supported
+ * by the server. We can't initialize it to null because then no lsp methods would be
+ * supported. Instead, we initialize it to a dummy instance which is then replaced.
+ *
+ * @param underlying
+ *   underlying instance which is swapped at runtime
+ */
+class DelegatingScalaService(
+    var underlying: ScalaLspService
+) extends ScalaLspService {
+
+  override def didOpen(
```

Comment:
There are only the ones implemented by Metals?

So now when adding a new functionality, you would need to add the method in 3 places? Could you add a comment about it in each of the classes we might need to add it, as noticable as possible? :sweat_smile: 

There was already a bit of issues with the DelegatingMetalsClient etc. (that is still needed, no?)

Reply:
Changed in [dff67d4](https://github.com/scalameta/metals/pull/4776/commits/dff67d4fb2d241a03bf340a62114eb44231a0a8c). Now I pass null as initial value to delegating service (still fine as it's not called until initialize and correct value is set before). I got rid of default implementations so now adding new endpoint in trait will cause compilation errors in both DelegatingService and in MetalsLspService (which are exactly 2 places which needs to be changed)